packages/flutter/lib/src/cupertino/theme.dart

```diff
@@ -54,7 +54,19 @@ class CupertinoTheme extends StatelessWidget {
   /// exist in the ancestry tree.
   static CupertinoThemeData of(BuildContext context) {
     final _InheritedCupertinoTheme inheritedTheme = context.inheritFromWidgetOfExactType(_InheritedCupertinoTheme);
-    return inheritedTheme?.theme?.data ?? const CupertinoThemeData();
+    return (inheritedTheme?.theme?.data ?? const CupertinoThemeData()).resolveFrom(context, nullOk: true);
```

Comment:
Shouldn't the inherited theme data be resolved against context?

Reply:
Not sure I follow this one. It is resolved against the `context`?

Reply:
context.inheritFromWidgetOfExactType(_InheritedCupertinoTheme) returns the closest _InheritedCupertinoTheme widget that's an ancestor of context. Its colors will have been resolved relative to the context that the _InheritedCupertinoTheme widget _itself_ was built in. So I think you'll want to resolve the colors against the local context parameter.

Reply:
I think I resolved the color in the return expression?

Reply:
Yes! Sorry if I missed that.